site/src/xServices/workspaceSchedule/workspaceScheduleXService.ts

```diff
@@ -50,160 +53,229 @@ export type WorkspaceScheduleEvent =
+  createMachine(
+    {
+      id: "workspaceScheduleState",
+      predictableActionArguments: true,
+      tsTypes: {} as import("./workspaceScheduleXService.typegen").Typegen0,
+      schema: {
+        context: {} as WorkspaceScheduleContext,
+        events: {} as WorkspaceScheduleEvent,
+        services: {} as {
+          getWorkspace: {
+            data: TypesGen.Workspace
+          }
+          getTemplate: {
+            data: TypesGen.Template
+          }
+        },
       },
-    },
-    initial: "idle",
-    on: {
-      GET_WORKSPACE: "gettingWorkspace",
-    },
-    states: {
-      idle: {
-        tags: "loading",
+      initial: "idle",
+      on: {
+        GET_WORKSPACE: "gettingWorkspace",
       },
-      gettingWorkspace: {
-        entry: ["clearGetWorkspaceError", "clearContext"],
-        invoke: {
-          src: "getWorkspace",
-          id: "getWorkspace",
-          onDone: {
-            target: "gettingPermissions",
-            actions: ["assignWorkspace"],
+      states: {
+        idle: {
+          tags: "loading",
+        },
+        gettingWorkspace: {
+          entry: ["clearGetWorkspaceError", "clearContext"],
+          invoke: {
+            src: "getWorkspace",
+            id: "getWorkspace",
+            onDone: {
+              target: "gettingPermissions",
+              actions: ["assignWorkspace"],
+            },
+            onError: {
+              target: "error",
+              actions: ["assignGetWorkspaceError"],
+            },
           },
-          onError: {
-            target: "error",
-            actions: ["assignGetWorkspaceError"],
+          tags: "loading",
+        },
+        gettingPermissions: {
```

Comment:
Permissions are loaded in the initial app loading in the auth machine so I don't think we need to load it again. If there are new permissions, I would add them there.

I would do is something like this:
```
const permissions = usePermissions()
const [workspaceScheduleState] = useMachine(workspaceSchedule, { context: { permissions } })
```

Reply:
Okay, I didn't add the permissions stuff in this PR (the machine just got reformatted), but I'll make a note to look into that later.